src/pages/CreateGroup/CreateGroup.tsx

```diff
@@ -0,0 +1,153 @@
+import Row from "../../components/layout/Row";
+import { Link, useNavigate } from "react-router-dom";
+import LeftArrowIcon from "../../assets/arrow_left.svg";
+import RemoveIcon from "../../assets/close.svg";
+import React, { useState } from "react";
+import TextField from "../../components/TextField";
+import Column from "../../components/layout/Column";
+import Button from "../../components/Button";
+import { SimpleUser, User } from "../../data/models/User";
+import { useApi } from "../../contexts/ApiProvider";
+import ConfirmationModal from "../../components/modals/ConfirmationModal";
+import { Group } from "../../data/models/Group";
+import toast from "react-hot-toast";
+import { useUser } from "../../contexts/UserProvider";
+
+export default function CreateGroup() {
+  const { api } = useApi();
+  const { user } = useUser();
+  const navigate = useNavigate();
+
+  const [groupName, setGroupName] = useState("");
+  const [curUserAdd, setCurUserAdd] = useState("");
+  const [groupUsers, setGroupUsers] = useState<SimpleUser[]>([]);
+
+  const [nameError, setNameError] = useState("");
+  const [userAddError, setUserAddError] = useState("");
+
+  // Create group, invite users, navigate to group, and show toast
+  const createGroup = async () => {
+    const newGroup = await api.post<Group>("/group/", {
+      name: groupName,
+    });
+    await api.put<Group>("/group/" + newGroup.data.id, {
+      invite_users: groupUsers.map((u) => u.username),
+    });
```

Comment:
You should be using `useMutation` to handle mutations like this. It will give you access to `onSuccess` and `onError` methods, and also just for code base consistency. Like here, if creating a group fails we don't want to navigate to group page, so we should only have the navigation and show the toast in `onSuccess`.

Reply:
as if right now if the request fails, the api client throws an error exception, causing the statements after (navigation and toast) to not be executed